pytest_doctestplus/plugin.py

```diff
@@ -192,22 +203,36 @@ def parse(self, s, name=None):
             skip_next = False
             skip_all = False
 
+            comment_char = '..'
             for entry in result:
                 if isinstance(entry, six.string_types) and entry:
+                    file_format = config.getoption('text_file_format', None)
+
+                    if file_format in comment_characters:
+                        comment_char = comment_characters[file_format]
+
                     required = []
                     skip_next = False
                     lines = entry.strip().splitlines()
-
-                    if '.. doctest-skip-all' in (x.strip() for x in lines):
+                    if '{} doctest-skip-all'.format(comment_char) in (x.strip() for x in lines):
                         skip_all = True
                         continue
 
                     if not len(lines):
                         continue
 
-                    last_line = lines[-1]
-                    match = re.match(
-                        r'\.\.\s+doctest-skip\s*::(\s+.*)?', last_line)
+                    # We allow last and second to last lines to match to allow
+                    # special environment to be in between, e.g. \begin{python}
+                    last_lines = lines[-2:]
+                    matches = [re.match(
+                        r'{}\s+doctest-skip\s*::(\s+.*)?'.format(comment_char),
+                        last_line) for last_line in last_lines]
+
+                    if len(matches) > 1:
+                        match = matches[0] or matches[1]
+                    else:
+                        match = matches[0]
```

Comment:
What is the purpose for this new logic? I wonder if we just need to tweak the regex a bit to avoid having to do this.

Reply:
Sometimes there is only a single line of text between two examples resulting an IndexError. 

I had hard times to make this work with multi like regex, so rather just gave up and kept the line by line iteration. I agree that this is a hack, but an even more complex regex would just another hack, too.